File: SDE_sheet/Arrays/04_kadane_algorithm/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        maxim = float('-inf')
        curr = 0
        
        for num in nums:
            curr = max(curr + num, num)
            maxim = max(maxim, curr)
        
        return maxim

File: SDE_sheet/Arrays/04_kadane_algorithm/solution.go
func maxSubArray(nums []int) int {
    maxim := math.MinInt32
    curr := 0
    
    for _, num := range nums {
        curr = max(curr + num, num)
        maxim = max(maxim, curr)
    }
    
    return maxim
}

func max(x, y int) int {
    if x > y {
        return x
    }
    return y
}
